### PR TITLE
OrgU/LUI: remove ilAdvandedSelectionListGUI, ilLinkButton

### PR DESCRIPTION
**Pull request by nhaagen**

### CODE REVIEW

File: Modules/OrgUnit/classes/LocalUser/class.ilLocalUserGUI.php [hunk 1]
```diff
@@ -57,13 +59,19 @@ public function __construct(ilObjectGUI $parentGui)
...
 
         $this->lng->loadLanguageModule('user');
         if (!$this->rbacSystem->checkAccess("cat_administrate_users", $this->parentGui->getObject()->getRefId())) {
             $this->tpl->setOnScreenMessage('failure', $this->lng->txt("msg_no_perm_admin_users"), true);
         }
     }
 
+    public function getRefId()
+    {
+        return $_GET['ref_id'];
```
Comment (klees) [inline Modules/OrgUnit/classes/LocalUser/class.ilLocalUserGUI.php]: Please use the [request wrapper](https://github.com/ILIAS-eLearning/ILIAS/blob/release_8/src/HTTP/Wrapper/WrapperFactory.php#LL38C22-L38C22) here.

File: Modules/OrgUnit/classes/class.ilOrgUnitLocalDIC.php [hunk 2]
```diff
@@ -53,7 +55,21 @@ protected static function buildDIC(): Container
             $dic["repo.Operations"],
             $dic["repo.OperationContexts"]
         );
+        $dic['ui.factory'] = fn (): \ILIAS\UI\Factory => $DIC['ui.factory'];
```
Comment (klees) [inline Modules/OrgUnit/classes/class.ilOrgUnitLocalDIC.php]: Please use `static fn` whenever applicable.

File: Modules/OrgUnit/classes/ARHelper/RowActions.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\Modules\OrgUnit\ARHelper;
+
+class RowActions
```
Comment (klees) [inline Modules/OrgUnit/classes/ARHelper/RowActions.php]: Please rename to "DropdownBuilder" and add a little context in the comment. I guess we would want to have an interface like this one on the `Dropdown` itself. Please propose an according entry to the roadmap of the UI framework in a separate PR.

File: Modules/OrgUnit/classes/ARHelper/RowActions.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        $clone = clone $this;
+        $clone->items[] = $this->ui_f->button()->shy(
+            $items[] = $this->lng->txt($label_lang_var),
+            $url
+        );
+        return $clone;
+    }
+
+    public function reset(): self
```
Comment (klees) [inline Modules/OrgUnit/classes/ARHelper/RowActions.php]: Remove `reset` and instead just construct a new builder whenever you need one.

File: Modules/OrgUnit/classes/Staff/class.ilOrgUnitStaffGUI.php [hunk 5]
```diff
@@ -1,414 +0,0 @@
...
-
-/**
- * Class ilOrgUnitStaffGUI
- * @author            Oskar Truffer <ot@studer-raimann.ch>
- * @author            Martin Studer <ms@studer-raimann.ch>
- * Date: 4/07/13
- * Time: 1:09 PM
- * @ilCtrl_Calls      ilOrgUnitStaffGUI: ilRepositorySearchGUI
- */
-class ilOrgUnitStaffGUI
```
Comment (klees) [inline Modules/OrgUnit/classes/Staff/class.ilOrgUnitStaffGUI.php:28]: I don't think you can delete these. They seemed to be [called from `ilObjOrgUnitGUI`](https://github.com/ILIAS-eLearning/ILIAS/blob/release_8/Modules/OrgUnit/classes/class.ilObjOrgUnitGUI.php).

**Review by klees — CHANGES_REQUESTED** [review]
Hi @nhaagen,

LGTM mostly, but some minor requests.

Thanks!

File: Modules/OrgUnit/classes/Staff/class.ilOrgUnitStaffGUI.php (hunk 5 above)
Comment (nhaagen) [reply Modules/OrgUnit/classes/Staff/class.ilOrgUnitStaffGUI.php:28]: I looked at the (removed) GUIs - they double ilOrgUnitUserAssignmentGUI and I'm quite sure they are not being called anymore

**Pull request merged**